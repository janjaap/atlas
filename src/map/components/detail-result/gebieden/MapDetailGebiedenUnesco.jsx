import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultWrapper from '../MapDetailResultWrapper';

const MapDetailGebiedenUnesco = ({ panoUrl, unesco, onMaximize, onPanoPreviewClick }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={unesco.label}
    title="UNESCO"
  />
);

MapDetailGebiedenUnesco.propTypes = {
  unesco: PropTypes.shape({
    label: PropTypes.string.isRequired
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailGebiedenUnesco;
