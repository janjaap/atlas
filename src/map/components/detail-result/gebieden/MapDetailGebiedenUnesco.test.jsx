import React from 'react';
import { shallow } from 'enzyme';

import MapDetailGebiedenUnesco from './MapDetailGebiedenUnesco';

describe('MapDetailGebiedenUnesco', () => {
  it('should render everything', () => {
    const unesco = {
      label: 'Unesco label'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailGebiedenUnesco
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        unesco={unesco}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
