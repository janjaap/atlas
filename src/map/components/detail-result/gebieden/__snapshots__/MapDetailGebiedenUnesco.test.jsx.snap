// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailGebiedenUnesco should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Unesco label"
  title="UNESCO"
/>
`;
